src/xtgeo/surface/regular_surface.py

```diff
@@ -150,56 +146,153 @@ def surface_from_grid3d(grid, template=None, where="top", mode="depth", rfactor=
 
     .. versionadded:: 2.1
     """
-    obj = RegularSurface()
-
+    obj = RegularSurface(1, 1, 0.0, 0.0)  # <- unimportant as it gets reset
     obj.from_grid3d(grid, template=template, where=where, mode=mode, rfactor=rfactor)
 
     return obj
 
 
-# ======================================================================================
-# RegularSurface class:
+def _data_reader_factory(file_format):
+    if file_format == "irap_binary":
+        return _regsurf_import.import_irap_binary
+    if file_format == "irap_ascii":
+        return _regsurf_import.import_irap_ascii
+    if file_format == "ijxyz":
+        return _regsurf_import.import_ijxyz
+    if file_format == "petromod":
+        return _regsurf_import.import_petromod
+    if file_format == "zmap_ascii":
+        return _regsurf_import.import_zmap_ascii
+    if file_format == "xtg":
+        return _regsurf_import.import_xtg
+    if file_format == "hdf5_regsurf":
+        return _regsurf_import.import_hdf5_regsurf
+    raise ValueError(f"Unknown file format {file_format}")
+
+
+def allow_deprecated_init(func):
+    # This decorator is here to maintain backwards compatibility in the construction
+    # of RegularSurface and should be deleted once the deprecation period has expired,
+    # the construction will then follow the new pattern.
+    @functools.wraps(func)
+    def wrapper(cls, *args, **kwargs):
+        # Checking if we are doing an initialization
+        # from file and raise a deprecation warning if
+        # we are.
+        if "sfile" in kwargs or len(args) == 1:
+            warnings.warn(
+                "Initializing directly from file name is deprecated and will be "
+                "removed in xtgeo version 4.0. Use the corresponding class method for "
+                'the file type instead, or RegularSurface.read_file("file_name")',
+                DeprecationWarning,
+            )
```

Comment:
Should it be: ... instead, or RegularSurface**()**.read_file("file_name")'?

Reply:
Both are valid, but `RegularSurface.read_file("file_name")` is only available for a `classmethod`, so to signify that it is a class method, I personally prefer that for clarity.